core/src/test/java/org/apache/brooklyn/enricher/stock/TransformingEnricherTest.java

```diff
@@ -225,17 +226,38 @@ public void testInfersTargetSensorSameAsSource() throws Exception {
         EntityAsserts.assertAttributeEqualsEventually(app, intSensorA, 3);
     }
 
+    @Test
+    public void testAllowCyclicPublishing() throws Exception {
+        app.enrichers().add(EnricherSpec.create(Transformer.class)
+                .configure(Transformer.SOURCE_SENSOR, intSensorA)
+                .configure(Transformer.TARGET_SENSOR, intSensorA)
+                .configure(Transformer.ALLOW_CYCLIC_PUBLISHING, true)
+                .configure(Transformer.TRANSFORMATION_FROM_VALUE, new Function<Integer, Object>() {
+                    @Override public Object apply(Integer input) {
+                        if (input != null && input < 10) {
+                            return input + 1;
+                        } else {
+                            return Entities.UNCHANGED;
+                        }
+                    }}));
+
+        app.sensors().set(intSensorA, 3);
+
+        EntityAsserts.assertAttributeEqualsEventually(app, intSensorA, 10);
```

Comment:
Follow up with `EntityAsserts.assertAttributeEqualsContinually` to ensure value remains at 10